diesel/src/mysql/connection/bind.rs

```diff
@@ -40,6 +71,56 @@ impl Binds {
         &mut self.mysql_binds
     }
 
+    /// The caller of this function must immediately check for errors
+    /// after return
+    pub fn populate_dynamic_buffers(&mut self, stmt: *mut ffi::MYSQL_STMT) {
+        use std::cmp::min;
+
+        for (i, buffer) in self.data.iter_mut().enumerate() {
+            if self.errors.as_ref().unwrap()[i] != 0 {
+                let written_bytes = buffer.len();
+                let needed_capacity = self.lengths[i] as usize;
+                buffer.reserve(needed_capacity - written_bytes);
```

Comment:
The only time we'd be in this branch where that wasn't true was a bug in the MySQL C library. Since there's no memory safety violation if that bug occurred (it'd probably just abort with OOM), I don't think we need to defend against it here.